lib/Module/IntrinsicCleaner.cpp

```diff
@@ -110,8 +114,12 @@ bool IntrinsicCleanerPass::runOnBasicBlock(BasicBlock &b, Module &M) {
       case Intrinsic::uadd_with_overflow:
       case Intrinsic::usub_with_overflow:
       case Intrinsic::umul_with_overflow: {
+#if LLVM_VERSION_CODE >= LLVM_VERSION(3, 8)
+        // ctor needs the iterator, but we already increased ours
+        IRBuilder<> builder(ii->getParent(), i_);
```

Comment:
Am I missing something, but I can't see why this would be necessary in 3.8, the constructros doesn't seem to have changed?